catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/BitReader.java

```diff
@@ -0,0 +1,191 @@
+        if (currentByte == null) {
+            if (byteBuf.readableBytes() < 1) {
+                throw new EOFException("read past end-of-file");
+            }
+            currentByte = byteBuf.readByte();
+            currentIndex = MAX_BIT_INDEX;
+        }
+
+        byte result = (byte) ((currentByte & (1 << currentIndex)) >> currentIndex);
+
+        return result == 0b1;
+    }
+
+    /**
+     * Get the number of bits that can be read.
+     *
+     * @return number of readable bits
+     */
+    public long readableBits() {
+        return byteBuf.readableBytes() * 8 + (currentByte != null ? currentIndex + 1 : 0);
+    }
+
+    /**
+     * Only call this method if we are at a byte boundary. Negative values are handled correctly.
+     *
+     * @param numberOfBytes the number of bytes to skip
+     */
+    private void skipBytes(int numberOfBytes) {
+        int remaining = numberOfBytes;
+        if (remaining > 0 && currentByte != null) {
+            currentByte = null;
+            remaining--;
+        }
+        if (remaining > 0) {
+            byteBuf.skipBytes(remaining);
+        }
+    }
+
+    public void skipBits(int numberOfBits) throws EOFException {
+
+        Validate.inclusiveBetween(0, Integer.MAX_VALUE, numberOfBits, "numberOfBits must >=0");
+
+        if (isByteBoundary() && numberOfBits % 8 == 0) {
+            skipBytes(numberOfBits / 8);
+            return;
+        }
+
+        int tmp = numberOfBits;
+        while (tmp >= 32) {
+            readBits(32);
+            tmp -= 32;
+        }
+        if (tmp > 0) {
+            readBits(tmp);
+        }
+    }
+
+    public byte readBit() throws EOFException {
+
+        byte result = testBit() ? (byte) 1 : (byte) 0;
+
+        consumeBit();
+
+        return result;
+    }
+
+    private void consumeBit() {
+        if (currentIndex > 0) {
+            currentIndex--;
+        } else if (currentIndex == 0) {
+            currentByte = null;
+        }
+    }
+
+    public long readBits(int numberOfBits) throws EOFException {
+
+        Validate.inclusiveBetween(1, 32, numberOfBits, "numberOfBits must be [1,32]");
+
+        if (numberOfBits == 1) {
+            return readBit();
+        }
+
+        long result = 0;
+
+        for (int i = 0; i < numberOfBits; i++) {
+            result |= readBit() << (numberOfBits - i - 1);
+        }
+
+        return result;
+    }
+
+    public boolean isByteBoundary() {
+        return currentByte == null || currentIndex == MAX_BIT_INDEX;
+    }
+
+    private void discardCurrentByte() {
+        currentByte = null;
+    }
+
+    public Optional<Long> findStart() {
+
+        if (!isByteBoundary()) {
+            discardCurrentByte();
+        }
+
+        int startPos = byteBuf.readerIndex();
+
+        startCode.init();
+
+        int position = byteBuf.forEachByte(startCode);
+
+        if (position == BYTE_BUF_EOF) {
+            return Optional.empty();
+        }
+
+        skipBytes(position - startPos - 3);
```

Comment:
Replaced 3 with a static final that describes the value, and improved the javadoc for StartCode.